examples/lines_bars_and_markers/bar_label_demo.py

```diff
@@ -0,0 +1,136 @@
+"""
+==============
+Bar Label Demo
+==============
+
+This example shows how to use the `~.Axes.bar_label` helper function
+to create bar chart labels.
+
+See also the :doc:`grouped bar
+</gallery/lines_bars_and_markers/barchart>`,
+:doc:`stacked bar
+</gallery/lines_bars_and_markers/bar_stacked>` and
+:doc:`horizontal bar chart
+</gallery/lines_bars_and_markers/barh>` examples.
+"""
+
+import matplotlib
+import matplotlib.pyplot as plt
+import numpy as np
+
+###############################################################################
+# Define the data
+
+N = 5
+menMeans = (20, 35, 30, 35, -27)
+womenMeans = (25, 32, 34, 20, -25)
+menStd = (2, 3, 4, 1, 2)
+womenStd = (3, 5, 2, 3, 3)
+ind = np.arange(N)    # the x locations for the groups
+width = 0.35       # the width of the bars: can also be len(x) sequence
+
+###############################################################################
+# Grouped bar chart
+
+fig1, ax1 = plt.subplots()
```

Comment:
I'd just go with
```suggestion
fig, ax = plt.subplots()
```
Since the example are in separate sections, they are semantically independent. Even though you are overwriting the variables because this is generated from one file, it's conceptually simpler for the user if the Axes and Figures are not numbered as long as there is only one per example plot.

Holds also for the following code.